pandas/tests/arrays/test_datetimelike.py

```diff
@@ -651,3 +651,51 @@ def test_array_interface(self, period_index):
         result = np.asarray(arr, dtype="S20")
         expected = np.asarray(arr).astype("S20")
         tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "array,casting_nats",
+    [
+        (
+            pd.TimedeltaIndex(["1 Day", "3 Hours", "NaT"])._data,
+            (pd.NaT, np.timedelta64("NaT", "ns")),
+        ),
+        (
+            pd.date_range("2000-01-01", periods=3, freq="D")._data,
+            (pd.NaT, np.datetime64("NaT", "ns")),
+        ),
+        (pd.period_range("2000-01-01", periods=3, freq="D")._data, (pd.NaT,)),
+    ],
+    ids=lambda x: type(x).__name__,
+)
+def test_casting_nat_setitem_array(array, casting_nats):
+    expected = type(array)._from_sequence([pd.NaT, array[1], array[2]])
+
+    for nat in casting_nats:
```

Comment:
can you make another level of parameterization to make these truly separate tests?

Reply:
I don't think so; the `casting_nats` parameter depends on the `array` parameter.

I've been looking at pytest extensions that might make this kind of thing possible